pytorch_lightning/callbacks/pt_callbacks.py

```diff
@@ -193,65 +197,91 @@ def __init__(self, filepath, monitor='val_loss', verbose=0,
 
         if mode == 'min':
             self.monitor_op = np.less
-            self.best = np.Inf
+            self.kth_value = np.Inf
+            self.mode = 'min'
         elif mode == 'max':
             self.monitor_op = np.greater
-            self.best = -np.Inf
+            self.kth_value = -np.Inf
+            self.mode = 'max'
         else:
             if 'acc' in self.monitor or self.monitor.startswith('fmeasure'):
                 self.monitor_op = np.greater
-                self.best = -np.Inf
+                self.kth_value = -np.Inf
+                self.mode = 'max'
             else:
                 self.monitor_op = np.less
-                self.best = np.Inf
+                self.kth_value = np.Inf
+                self.mode = 'min'
 
-    def save_model(self, filepath, overwrite):
-        dirpath = '/'.join(filepath.split('/')[:-1])
+    def _del_model(self, filepath):
+        dirpath = os.path.dirname(filepath)
 
         # make paths
-        os.makedirs(os.path.dirname(filepath), exist_ok=True)
+        os.makedirs(dirpath, exist_ok=True)
 
-        if overwrite:
-            for filename in os.listdir(dirpath):
-                if self.prefix in filename:
-                    path_to_delete = os.path.join(dirpath, filename)
-                    try:
-                        shutil.rmtree(path_to_delete)
-                    except OSError:
-                        os.remove(path_to_delete)
+        try:
+            shutil.rmtree(filepath)
+        except OSError:
+            os.remove(filepath)
+
+    def _save_model(self, filepath, overwrite):
+        dirpath = os.path.dirname(filepath)
+
+        # make paths
+        os.makedirs(dirpath, exist_ok=True)
 
         # delegate the saving to the model
         self.save_function(filepath)
 
+    def check_monitor_top_k(self, current):
+        return ((len(self.best_k_models.keys()) < self.save_top_k) or
```

Comment:
split these lines up for readability. I prefer to have boolean words such as:   
```python   
is_correct_fx = self.monitor_op(current, self.best_k_models[self.kth_value]))  
short_description = of complicated > expression % 0 == 2 or something crazy   
if is_correct_fx and short_description:
    # more readable

Reply:
@williamFalcon Do you prefer this one?

```python
less_than_k_models = len(self.best_k_models.keys()) < self.save_top_k
if less_than_k_models:
    return True
else:
    return self.monitor_op(current, self.best_k_models[self.kth_value])
```

Reply:
having multiple `return`s depending on single `if` is not very nice especially while debugging

Reply:
```